indent rule broken inside fat-arrow function

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 3.12.0
* **Node Version:** 6.2.0
* **npm Version:** 3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**
`babel-eslint`

**Please show your full configuration:**

<!-- Paste your configuration below: -->
**1**
```
{
  "rules": {
    "indent" : [2, 2, {"ObjectExpression": 1}]
  }
}
```
**2**
```
{
  "rules": {
    "indent" : [2, 2, {"ObjectExpression": "first"}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
    {
      display,
    }
  );
};
```

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
**1**
```bash
    4:5   error    Expected indentation of 23 spaces but found 4   indent
    5:7   error    Expected indentation of 25 spaces but found 6   indent
    6:5   error    Expected indentation of 23 spaces but found 4   indent
```
**2**
```bash
    4:5   error    Expected indentation of 23 spaces but found 4   indent
    6:5   error    Expected indentation of 23 spaces but found 4   indent
```

**The following code passes:**
**1**
```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
                       {
                         display,
                       }
  );
};
```
**2**
```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
                       {
      display,
                       }
  );
};
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -737,8 +737,12 @@ module.exports = {
                         } else if (parent.type === "CallExpression" || parent.type === "NewExpression") {
                             if (typeof options.CallExpression.arguments === "number") {
                                 nodeIndent += options.CallExpression.arguments * indentSize;
-                            } else if (parent.arguments.indexOf(node) !== -1) {
-                                nodeIndent = parent.arguments[0].loc.start.column;
+                            } else if (options.CallExpression.arguments === "first") {
+                                if (parent.arguments.indexOf(node) !== -1) {
+                                    nodeIndent = parent.arguments[0].loc.start.column;
+                                }
+                            } else {
+                                nodeIndent += indentSize;
                             }
                         } else if (parent.type === "LogicalExpression" || parent.type === "ArrowFunctionExpression") {
                             nodeIndent += indentSize;

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1863,6 +1863,50 @@ ruleTester.run("indent", rule, {
             "    [\n" +
             "    ]()",
             options: [4, {CallExpression: {arguments: "first"}, ArrayExpression: "first"}]
+        },
+
+        // https://github.com/eslint/eslint/issues/7732
+        {
+            code:
+            "const lambda = foo => {\n" +
+            "  Object.assign({},\n" +
+            "    filterName,\n" +
+            "    {\n" +
+            "      display\n" +
+            "    }\n" +
+            "  );" +
+            "}",
+            options: [2, {ObjectExpression: 1}],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code:
+            "const lambda = foo => {\n" +
+            "  Object.assign({},\n" +
+            "    filterName,\n" +
+            "    {\n" +
+            "      display\n" +
+            "    }\n" +
+            "  );" +
+            "}",
+            options: [2, {ObjectExpression: "first"}],
+            parserOptions: { ecmaVersion: 6 }
+        },
+
+        // https://github.com/eslint/eslint/issues/7733
+        {
+            code:
+            "var foo = function() {\n" +
+            "\twindow.foo('foo',\n" +
+            "\t\t{\n" +
+            "\t\t\tfoo: 'bar'," +
+            "\t\t\tbar: {\n" +
+            "\t\t\t\tfoo: 'bar'\n" +
+            "\t\t\t}\n" +
+            "\t\t}\n" +
+            "\t);\n" +
+            "}",
+            options: ["tab"]
         }
     ],
     invalid: [

